src/actions/index.test.js

```diff
@@ -89,4 +111,256 @@ describe('Actions', () => {
+    });
+  });
+
+  describe('changeNodeDescription', () => {
+    const id = 42;
+    const description = 'lorem ipsum';
+
+    beforeEach(() => {
+      changeNodeDescription(id, description)(dispatch);
+    });
+
+    it('calls dispatch with type CHANGE_NODE_DESCRIPTION and payload', () => {
+      expect(dispatch).toHaveBeenNthCalledWith(1, {
+        type: CHANGE_NODE_DESCRIPTION,
+        payload: { id, description },
+      });
+    });
+
+    it('calls dispatch persist action', () => {
+      expect(dispatch).toHaveBeenNthCalledWith(2, persistStateAction);
+    });
+  });
+
+  describe('changeNodePosition', () => {
+    const id = 42;
+    const x = 0;
+    const y = 0;
+
+    beforeEach(() => {
+      changeNodePosition(id, x, y)(dispatch);
+    });
+
+    it('calls dispatch with type CHANGE_NODE_POSITION and payload', () => {
+      expect(dispatch).toHaveBeenNthCalledWith(1, {
+        type: CHANGE_NODE_POSITION,
+        payload: { id, x, y },
+      });
+    });
+
+    it('calls dispatch persist action', () => {
+      expect(dispatch).toHaveBeenNthCalledWith(2, persistStateAction);
+    });
+  });
+
+
+  describe('addNode', () => {
+    const id = 42;
+    const states = ['fake-statuses'];
+    const position = 42;
+
+    beforeEach(() => {
+      addNode(id, states, position)(dispatch);
+    });
+
+    it('calls dispatch with type ADD_NODE and payload with payload', () => {
+      expect(dispatch).toHaveBeenNthCalledWith(1, {
+        type: ADD_NODE,
+        payload: { id, states, position },
+      });
+    });
+
+    it('calls dispatch persist action', () => {
+      expect(dispatch).toHaveBeenNthCalledWith(2, persistStateAction);
+    });
+  });
+
+  describe('removeNode', () => {
+    const id = 42;
+    const nodes = [];
+    const getState = () => nodes;
+
+    beforeEach(() => {
+      removeNode(id)(dispatch, getState);
+    });
+
+    it('calls dispatch with type REMOVE_NODE and payload with payload', () => {
+      expect(dispatch).toHaveBeenNthCalledWith(1, {
+        type: REMOVE_NODE,
+        payload: { id, nodes },
+      });
+    });
+
+    it('calls dispatch persist action', () => {
+      expect(dispatch).toHaveBeenNthCalledWith(2, persistStateAction);
+    });
+  });
+
+
+  describe('addSuperNode', () => {
+    const id = 42;
+    const name = 'name';
+    const position = 42;
+
+    describe('state with network id', () => {
+      const state = {
+        network: { id },
+      };
+
+      beforeEach(() => {
+        addSuperNode(state, position)(dispatch);
+      });
+
+      it('calls dispatch with type ADD_SUPER_NODE and payload with payload', () => {
+        expect(dispatch).toHaveBeenNthCalledWith(1, {
+          type: ADD_SUPER_NODE,
+          payload: { id, state, position },
+        });
+      });
+
+      it('calls dispatch persist action', () => {
+        expect(dispatch).toHaveBeenNthCalledWith(2, persistStateAction);
+      });
+    });
+
+    describe('state with network name', () => {
```

Comment:
Can you update this description to `When the network has no id`?
Because every network has `name`, but not always `id` ;D